components/post_view/post_attachment_container/post_attachment_container.tsx

```diff
@@ -7,33 +7,59 @@ import {useHistory} from 'react-router-dom';
 export type Props = {
     className?: string;
     children?: JSX.Element;
+    preventClickAction?: boolean;
     link: string;
 };
+
 const PostAttachmentContainer = (props: Props) => {
-    const {children, className, link} = props;
+    const {children, className, link, preventClickAction} = props;
     const history = useHistory();
-    const handleOnClick = useCallback((e) => {
-        const {tagName} = e.target;
-        e.stopPropagation();
-        const elements = ['A', 'IMG', 'BUTTON', 'I'];
+    const handleOnClick = useCallback(
+        (e) => {
+            const {tagName} = e.target;
+            e.stopPropagation();
+            const elements = ['A', 'IMG', 'BUTTON', 'I'];
 
-        if (!elements.includes(tagName) && (e.target.getAttribute('role') !== 'button' && e.target.className !== `attachment attachment--${className}`)) {
-            const classNames = ['icon icon-menu-down', 'icon icon-menu-right', 'post-image', 'file-icon'];
+            if (
+                !elements.includes(tagName) &&
+                e.target.getAttribute('role') !== 'button' &&
+                e.target.className !== `attachment attachment--${className}`
+            ) {
+                const classNames = [
+                    'icon icon-menu-down',
+                    'icon icon-menu-right',
+                    'post-image',
+                    'file-icon',
+                ];
 
-            if (!classNames.some((className) => e.target.className.includes(className)) && e.target.id !== 'image-name-text') {
-                history.push(link);
+                if (
+                    !classNames.some((className) =>
+                        e.target.className.includes(className),
+                    ) &&
+                    e.target.id !== 'image-name-text'
+                ) {
+                    history.push(link);
+                }
             }
-        }
-    }, [history]);
+        },
+        [history, preventClickAction],
```

Comment:
I don't think this actually depends on `preventClickAction` any more, so that can be removed